Gems/ROS2/Code/Source/Manipulation/JointStatePublisher.cpp

```diff
@@ -30,36 +31,40 @@ namespace ROS2
         rosHeader.stamp = ROS2::ROS2Interface::Get()->GetROSTimestamp();
         m_jointStateMsg.header = rosHeader;
 
+        for (size_t i = 0; i < m_jointNames.size(); i++)
```

Comment:
Good practice - iterate over container you write to instead of data source.
Add a check whether the joint names matches the message name vector size.